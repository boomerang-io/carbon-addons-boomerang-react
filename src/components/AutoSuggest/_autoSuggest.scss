@include exports('addons-bmrg-autoSuggest') {
  .#{$prefix}--bmrg-auto-suggest {
    width: 100%;

    .react-autosuggest__suggestions-container {
      background-color: $ui-01;
      box-shadow: 0 0.1875rem 0.1875rem 0 rgba(0, 0, 0, 0.1);
      color: $text-02;
      max-height: 8.75rem;
      overflow-y: auto;
      width: 100%;
    }

    .react-autosuggest__suggestion {
      cursor: pointer;
      font-size: 0.875rem;
      font-weight: 400;
      height: 2.5rem;
      letter-spacing: 0.01rem;
      line-height: 1rem;
      overflow: hidden;
      padding: 0.75rem 1rem;
      position: relative;
      text-overflow: ellipsis;
      white-space: nowrap;

      &:last-of-type {
        &:after {
          background-color: transparent;
        }
      }

      &:after {
        background-color: $ui-03;
        bottom: 0;
        content: '';
        height: 0.08rem;
        left: 1rem;
        position: absolute;
        width: calc(100% - 2rem);
      }

      &--highlighted {
        color: $interactive-02;
        background-color: $hover-ui;

        &:after {
          background-color: transparent;
        }
      }
    }
  }
}
