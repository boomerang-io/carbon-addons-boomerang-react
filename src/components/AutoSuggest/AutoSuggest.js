import React, { Component } from 'react';
import PropTypes from 'prop-types';
import ms from 'match-sorter';

import AutoSuggestInput from './AutoSuggestInput';

const SELECT_METHODS = ['up', 'down', 'click'];

class AutoSuggest extends Component {
  static propTypes = {
    autoSuggestions: PropTypes.array.isRequired,
    children: PropTypes.node,
    initialValue: PropTypes.string,
    inputProps: PropTypes.object,
    onChange: PropTypes.func,
  };

  static defaultProps = {
    onChange: () => {},
    initialValue: '',
  };

  inputRef = React.createRef();

  state = {
    // Used if we want to have some of the functions external instead of
    // the default ones in the wrapper
    value: this.props.initialValue,
    caretIndex: 0,
    suggestions: [],
  };

  // Each time the component updates we want to refocus the input and keep the cursor in the correct place
  // Needed for when cycling through mutliple suggestions with the arrow keys and the cursor resets to the end of the input. We don't want that.
  componentDidUpdate(prevProps, prevState) {
    if (this.state.value && prevState.value !== this.state.value) {
      // prevent it from focusing on initial render / empty
      this.inputRef.current.focus();
      this.inputRef.current.setSelectionRange(this.state.caretIndex, this.state.caretIndex);
    }
  }

  renderSuggestion = (suggestion) => <div>{suggestion.label}</div>;

  onSuggestionsFetchRequested = () => {
    this.setState(() => ({
      suggestions: this.getSuggestions(),
    }));
  };

  /**
   * More logic here for handling a user cycling through suggestions
   * Move the caret to the new suggestion location or use the reference to the DOM element.
   * Shift based on the change in length of the value b/c of different length suggestions
   */

  onInputChange = (event, { newValue, method }) => {
    this.setState((prevState) => ({
      value: newValue,
      caretIndex: SELECT_METHODS.includes(method)
        ? prevState.caretIndex + (newValue.length - prevState.value.length)
        : this.inputRef.current.selectionStart,
    }));
    this.props.onChange(newValue);
  };

  /**
   * Return the new value for the input
   * - Find the current caret position
   * - get the string up to that point
   * - find the last word (space-delimited) and replace it in input
   * -
   */
  getSuggestionValue = (suggestion) => {
    const inputWords = this.findWordsBeforeCurrentLocation();
    // const propertySuggestion = `\${p:${suggestion}}`;

    /*
     * Find the position of the caret, get the string up to that point
     * and find the index of the last word - i.e. the one the user entered
     */
    const pos = this.state.value
      .slice(0, this.state.caretIndex)
      .lastIndexOf(inputWords[inputWords.length - 1]);

    // Sub in the new property suggestion
    return (
      this.state.value.substring(0, pos) +
      suggestion.value +
      this.state.value.substring(pos + inputWords[inputWords.length - 1].length)
    );
  };

  getSuggestions = () => {
    const inputWords = this.findWordsBeforeCurrentLocation();
    // Prevent empty string from matching everyhing
    const inputWord = inputWords.length ? inputWords[inputWords.length - 1] : '';
    return !inputWord
      ? []
      : ms(this.props.autoSuggestions, inputWord, {
          // Use match-sorter for matching inputs
          keys: [{ key: 'value', threshold: ms.rankings.SIMPLE }],
        });
  };

  // Get array of distinct words prior to the current location of entered text
  // Use the inputRef instead of state becuase of asnychronous updating of state and calling of these functions :(
  findWordsBeforeCurrentLocation() {
    return this.inputRef.current.value.slice(0, this.inputRef.current.selectionStart).split(' ');
  }

  render() {
    const { inputProps, children, ...rest } = this.props;
    const finalInputProps = {
      onChange: this.onInputChange,
      value: this.state.value,
      ...inputProps,
    };
    return (
      <AutoSuggestInput
        getSuggestionValue={this.getSuggestionValue}
        inputProps={finalInputProps}
        onSuggestionsFetchRequested={this.onSuggestionsFetchRequested}
        renderSuggestion={this.renderSuggestion}
        suggestions={this.state.suggestions}
        {...rest}
      >
        {(inputProps) => React.cloneElement(children, { ...inputProps, ref: this.inputRef })}
      </AutoSuggestInput>
    );
  }
}

export default AutoSuggest;
