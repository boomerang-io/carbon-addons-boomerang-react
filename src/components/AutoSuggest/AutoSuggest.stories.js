import React from 'react';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { text } from '@storybook/addon-knobs/react';

import TextInput from '../TextInput';

import AutoSuggest from './AutoSuggest';

const animals = [
  { label: 'Caribou', value: 'caribou' },
  { label: 'Cat', value: 'cat' },
  { label: 'Catfish', value: 'catfish' },
  { label: 'Cheetah', value: 'cheetah' },
  { label: 'Chipmunk', value: 'chipmunk' },
  { label: 'Dog', value: 'dog' },
  { label: 'Dolphin', value: 'dolphin' },
  { label: 'Dove', value: 'dove' },
  { label: 'Panda', value: 'panda' },
  { label: 'Parrot', value: 'parrot' },
  { label: 'Peacock', value: 'peacock' },
  { label: 'Penguim', value: 'penguim' },
];

storiesOf('AutoSuggest', module).add('default', () => {
  return (
    <div style={{ width: '25rem' }}>
      <AutoSuggest
        autoSuggestions={animals}
        inputProps={{
          id: text('id', 'auto-suggest'),
          placeholder: text('placeholder', 'Type an animal'),
          labelText: text('labelText', 'Animal'),
        }}
        onChange={action('Auto suggest change')}
      >
        <TextInput />
      </AutoSuggest>
    </div>
  );
});
