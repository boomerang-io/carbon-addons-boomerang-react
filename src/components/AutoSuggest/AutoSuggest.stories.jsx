import { action } from "@storybook/addon-actions";
import { text } from "@storybook/addon-knobs";
import TextInput from "../TextInput";
import AutoSuggest from "./AutoSuggest";

export default {
  title: "Inputs/AutoSuggest",
  component: AutoSuggest,
};

const animals = [
  { label: "caribou", value: "caribou" },
  { label: "cat", value: "cat" },
  { label: "catfish", value: "catfish" },
  { label: "cheetah", value: "cheetah" },
  { label: "chipmunk", value: "chipmunk" },
  { label: "dog", value: "dog" },
  { label: "dolphin", value: "dolphin" },
  { label: "dove", value: "dove" },
  { label: "panda", value: "panda" },
  { label: "parrot", value: "parrot" },
  { label: "peacock", value: "peacock" },
  { label: "penguin", value: "penguin" },
];

export const Default = () => {
  return (
    <div style={{ width: "25rem" }}>
      <AutoSuggest
        autoSuggestions={animals}
        inputProps={{
          id: text("id", "auto-suggest"),
          placeholder: text("placeholder", "Type an animal"),
          labelText: text("labelText", "Animal"),
        }}
        onChange={action("Auto suggest change")}
      >
        <TextInput />
      </AutoSuggest>
    </div>
  );
};
