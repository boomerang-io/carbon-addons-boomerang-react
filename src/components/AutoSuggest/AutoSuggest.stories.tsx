import { action } from "@storybook/addon-actions";
import TextInput from "../TextInput";
import AutoSuggest from "./AutoSuggest";

const animals = [
  { label: "caribou", value: "caribou" },
  { label: "cat", value: "cat" },
  { label: "catfish", value: "catfish" },
  { label: "cheetah", value: "cheetah" },
  { label: "chipmunk", value: "chipmunk" },
  { label: "dog", value: "dog" },
  { label: "dolphin", value: "dolphin" },
  { label: "dove", value: "dove" },
  { label: "panda", value: "panda" },
  { label: "parrot", value: "parrot" },
  { label: "peacock", value: "peacock" },
  { label: "penguin", value: "penguin" },
];

export default {
  title: "Inputs/AutoSuggest",
  component: AutoSuggest,
  parameters: {
    docs: {
      description: {
        component: "An enhanced TextInput that supports selecting from a provided list of options.",
      },
    },
  },
  decorators: [(story: any) => <div style={{ maxWidth: "25rem", minHeight: "20rem" }}>{story()}</div>],
  argTypes: {
    autoSuggestions: { control: "array", defaultValue: animals },
    inputProps: {
      control: "object",
      defaultValue: {
        id: "auto-suggest",
        placeholder: "Type an animal",
        labelText: "Animal",
      },
    },
  },
};

export const Default = (args: any) => {
  return (
    <AutoSuggest onChange={action("Auto suggest change")} {...args}>
      <TextInput id="auto-suggest"/>
    </AutoSuggest>
  );
};
