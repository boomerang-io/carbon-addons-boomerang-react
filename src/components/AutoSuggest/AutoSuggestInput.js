import React, { Component } from 'react';
import PropTypes from 'prop-types';
import AutoSuggest from 'react-autosuggest';
import { settings } from 'carbon-components';

const { prefix } = settings;

const getSuggestions = (values, value) => {
  const inputValue = value.trim().toLowerCase();
  const inputLength = inputValue.length;

  return inputLength === 0
    ? []
    : values.filter((option) => option.toLowerCase().slice(0, inputLength) === inputValue);
};

class AutoSuggestInput extends Component {
  static propTypes = {
    autoSuggestions: PropTypes.array,
    children: PropTypes.node,
    getSuggestions: PropTypes.func,
    getSuggestionValue: PropTypes.func,
    inputProps: PropTypes.object,
    renderInputComponent: PropTypes.node,
    renderSuggestion: PropTypes.func,
    suggestions: PropTypes.array,
  };

  static defaultProps = {
    getSuggestions,
  };

  state = {
    suggestions: this.props.suggestions || [],
  };

  // allow external control of suggestions
  componentDidUpdate(prevProps) {
    const { suggestions } = this.props;
    if (this.didSuggestionsChange(prevProps.suggestions, suggestions)) {
      this.setState({ suggestions });
    }
  }

  didSuggestionsChange(suggestions1, suggestions2) {
    return JSON.stringify(suggestions1) !== JSON.stringify(suggestions2);
  }

  onSuggestionsFetchRequested = ({ value }) => {
    this.setState({
      suggestions: this.props.getSuggestions(this.props.autoSuggestions, value),
    });
  };

  onSuggestionsClearRequested = () => {
    this.setState({
      suggestions: [],
    });
  };

  render() {
    const {
      children,
      getSuggestionValue,
      inputProps,
      renderInputComponent,
      renderSuggestion,
      ...rest
    } = this.props;
    const { suggestions } = this.state;

    return (
      <div className={`${prefix}--bmrg-auto-suggest`}>
        <AutoSuggest
          getSuggestionValue={getSuggestionValue}
          inputProps={inputProps}
          onSuggestionsClearRequested={this.onSuggestionsClearRequested}
          onSuggestionsFetchRequested={this.onSuggestionsFetchRequested}
          renderInputComponent={
            children ? (inputProps) => children(inputProps) : renderInputComponent || undefined
          }
          renderSuggestion={renderSuggestion}
          suggestions={suggestions}
          {...rest}
        />
      </div>
    );
  }
}

export default AutoSuggestInput;
