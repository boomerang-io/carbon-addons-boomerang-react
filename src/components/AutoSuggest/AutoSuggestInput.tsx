import React, { Component } from "react";
// @ts-expect-error TS(7016): Could not find a declaration file for module 'reac... Remove this comment to see the full error message
import AutoSuggest from "react-autosuggest";
import { prefix } from "../../internal/settings";

const getSuggestions = (values: any, value: any) => {
  const inputValue = value.trim().toLowerCase();
  const inputLength = inputValue.length;

  return inputLength === 0 ? [] : values.filter((option: any) => option.toLowerCase().slice(0, inputLength) === inputValue);
};

type OwnAutoSuggestInputProps = {
    autoSuggestions?: any[];
    getSuggestions?: (...args: any[]) => any;
    getSuggestionValue?: (...args: any[]) => any;
    inputProps?: any;
    renderInputComponent?: React.ReactNode;
    renderSuggestion?: (...args: any[]) => any;
    suggestions?: any[];
};

type AutoSuggestInputState = any;

type AutoSuggestInputProps = OwnAutoSuggestInputProps & typeof AutoSuggestInput.defaultProps;

class AutoSuggestInput extends Component<AutoSuggestInputProps, AutoSuggestInputState> {

  static defaultProps = {
    getSuggestions,
  };

  state = {
    suggestions: this.props.suggestions || [],
  };

  // allow external control of suggestions
  componentDidUpdate(prevProps: AutoSuggestInputProps) {
    const { suggestions } = this.props;
    if (this.didSuggestionsChange(prevProps.suggestions, suggestions)) {
      this.setState({ suggestions });
    }
  }

  didSuggestionsChange(suggestions1: any, suggestions2: any) {
    return JSON.stringify(suggestions1) !== JSON.stringify(suggestions2);
  }

  onSuggestionsFetchRequested = ({
    value
  }: any) => {
    this.setState({
      suggestions: this.props.getSuggestions(this.props.autoSuggestions, value),
    });
  };

  onSuggestionsClearRequested = () => {
    this.setState({
      suggestions: [],
    });
  };

  render() {
    const { children, getSuggestionValue, inputProps, renderInputComponent, renderSuggestion, ...rest } = this.props;
    const { suggestions } = this.state;

    return (
      <div className={`${prefix}--bmrg-auto-suggest`}>
        <AutoSuggest
          getSuggestionValue={getSuggestionValue}
          inputProps={inputProps}
          onSuggestionsClearRequested={this.onSuggestionsClearRequested}
          onSuggestionsFetchRequested={this.onSuggestionsFetchRequested}
          // @ts-expect-error TS(2349): This expression is not callable.
          renderInputComponent={children ? (inputProps: any) => children(inputProps) : renderInputComponent || undefined}
          renderSuggestion={renderSuggestion}
          suggestions={suggestions}
          {...rest}
        />
      </div>
    );
  }
}

export default AutoSuggestInput;
