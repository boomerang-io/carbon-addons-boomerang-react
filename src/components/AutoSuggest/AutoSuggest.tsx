import React, { Component } from "react";
import { matchSorter as ms } from "match-sorter";

import AutoSuggestInput from "./AutoSuggestInput";

const SELECT_METHODS = ["up", "down", "click"];

type OwnProps = {
  autoSuggestions: any[];
  initialValue?: string;
  inputProps?: any;
  onChange?: (...args: any[]) => any;
};

type State = any;

type Props = OwnProps & typeof AutoSuggest.defaultProps;

class AutoSuggest extends Component<Props, State> {
  static defaultProps = {
    onChange: () => {},
    initialValue: "",
  };

  inputRef = React.createRef();

  state = {
    // Used if we want to have some of the functions external instead of
    // the default ones in the wrapper
    value: this.props.initialValue,
    caretIndex: 0,
    suggestions: [],
  };

  // Each time the component updates we want to refocus the input and keep the cursor in the correct place
  // Needed for when cycling through mutliple suggestions with the arrow keys and the cursor resets to the end of the input. We don't want that.
  componentDidUpdate(prevProps: Props, prevState: State) {
    if (this.state.value && prevState.value !== this.state.value) {
      // prevent it from focusing on initial render / empty
      (this as any).inputRef.current?.focus();
      (this as any).inputRef.current?.setSelectionRange(this.state.caretIndex, this.state.caretIndex);
    }
  }

  renderSuggestion = (suggestion: any) => <div>{suggestion.label}</div>;

  onSuggestionsFetchRequested = () => {
    this.setState(() => ({
      suggestions: this.getSuggestions(),
    }));
  };

  /**
   * More logic here for handling a user cycling through suggestions
   * Move the caret to the new suggestion location or use the reference to the DOM element.
   * Shift based on the change in length of the value b/c of different length suggestions
   */

  onInputChange = (event: any, { newValue, method }: any) => {
    this.setState((prevState: any) => ({
      value: newValue,
      caretIndex: SELECT_METHODS.includes(method)
        ? prevState.caretIndex + (newValue.length - prevState.value.length)
        : (this as any).inputRef.current.selectionStart,
    }));
    this.props.onChange(newValue);
  };

  /**
   * Return the new value for the input
   * - Find the current caret position
   * - get the string up to that point
   * - find the last word (space-delimited) and replace it in input
   * -
   */
  getSuggestionValue = (suggestion: any) => {
    const inputWords = this.findWordsBeforeCurrentLocation();
    // const propertySuggestion = `\${p:${suggestion}}`;

    /*
     * Find the position of the caret, get the string up to that point
     * and find the index of the last word - i.e. the one the user entered
     */
    const pos = this.state.value.slice(0, this.state.caretIndex).lastIndexOf(inputWords[inputWords.length - 1]);

    // Sub in the new property suggestion
    return (
      this.state.value.substring(0, pos) +
      suggestion.value +
      this.state.value.substring(pos + inputWords[inputWords.length - 1].length)
    );
  };

  getSuggestions = () => {
    const inputWords = this.findWordsBeforeCurrentLocation();
    // Prevent empty string from matching everyhing
    const inputWord = inputWords.length ? inputWords[inputWords.length - 1] : "";
    return !inputWord
      ? []
      : ms(this.props.autoSuggestions, inputWord, {
          // Use match-sorter for matching inputs
          keys: [{ key: "value", threshold: (ms.rankings as any).SIMPLE }],
        });
  };

  // Get array of distinct words prior to the current location of entered text
  // Use the inputRef instead of state becuase of asnychronous updating of state and calling of these functions :(
  findWordsBeforeCurrentLocation() {
    return (this as any).inputRef.current.value.slice(0, (this as any).inputRef.current.selectionStart).split(" ");
  }

  render() {
    const { inputProps, children, ...rest } = this.props;
    const finalInputProps = {
      onChange: this.onInputChange,
      value: this.state.value,
      ...inputProps,
    };
    return (
      <AutoSuggestInput
        getSuggestionValue={this.getSuggestionValue}
        inputProps={finalInputProps}
        // @ts-expect-error TS(2769): No overload matches this call.
        onSuggestionsFetchRequested={this.onSuggestionsFetchRequested}
        renderSuggestion={this.renderSuggestion}
        suggestions={this.state.suggestions}
        focusInputOnSuggestionClick={false}
        {...rest}
      >
        {/* @ts-expect-error TS(2769): No overload matches this call. */}
        {(inputProps: any) => React.cloneElement(children, { ...inputProps, ref: this.inputRef })}
      </AutoSuggestInput>
    );
  }
}

export default AutoSuggest;
