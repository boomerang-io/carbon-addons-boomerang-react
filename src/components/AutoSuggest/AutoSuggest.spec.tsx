/*
IBM Confidential
694970X, 69497O0
Â© Copyright IBM Corp. 2022, 2024
*/


import React from "react";
import { expect, test } from "vitest";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { axe } from "jest-axe";
import AutoSuggest from "./AutoSuggest";
import TextInput from "../TextInput";
import { animals } from "./AutoSuggest.stories";

const props = {
  autoSuggestions: animals,
  children: <TextInput id="suggestions" labelText="Suggestions" />,
  inputProps: {},
  onChange: () => {},
};

describe("AutoSuggest", () => {
  test("snapshot", () => {
    const { baseElement } = render(<AutoSuggest {...props} />);
    expect(baseElement).toMatchSnapshot();
  });

  test("functional", async () => {
    render(<AutoSuggest {...props} />);
    expect(await screen.findByLabelText("Suggestions")).toBeInTheDocument();
    const input = screen.getByLabelText("Suggestions");
    userEvent.type(input, "ca");
    expect(await screen.findByText("caribou")).toBeInTheDocument();
    expect(await screen.findByText("cat")).toBeInTheDocument();
    const suggestion = screen.getByText("caribou");
    await userEvent.click(suggestion);
    expect(screen.queryByText("cat")).toBeNull();
  });

  test("a11y", async () => {
    const { container } = render(<AutoSuggest {...props} />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});
