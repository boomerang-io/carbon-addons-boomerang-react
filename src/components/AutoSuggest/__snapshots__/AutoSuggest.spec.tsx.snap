// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AutoSuggest > snapshot 1`] = `
<body>
  <div>
    <div
      class="cds--bmrg-auto-suggest"
    >
      <div
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-owns="react-autowhatever-1"
        class="react-autosuggest__container"
        role="combobox"
      >
        <div
          class="cds--bmrg-text-input"
        >
          <div
            class="cds--form-item react-autosuggest__input cds--text-input-wrapper"
          >
            <div
              class="cds--text-input__label-wrapper"
            >
              <label
                class="cds--label"
                dir="auto"
                for="suggestions"
              >
                <div
                  class="cds--bmrg-text-input__label"
                >
                  <div>
                    Suggestions
                  </div>
                </div>
              </label>
            </div>
            <div
              class="cds--text-input__field-outer-wrapper"
            >
              <div
                class="cds--text-input__field-wrapper"
              >
                <input
                  aria-autocomplete="list"
                  aria-controls="react-autowhatever-1"
                  autocomplete="off"
                  class="cds--text-input"
                  control="object"
                  id="suggestions"
                  type="text"
                  value=""
                />
                
                <span
                  aria-atomic="true"
                  aria-live="assertive"
                  class="cds--text-input__counter-alert"
                  role="alert"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          aria-label="AutoSuggest listbox"
          class="react-autosuggest__suggestions-container"
          id="react-autowhatever-1"
          query=""
          role="listbox"
        />
      </div>
    </div>
  </div>
</body>
`;
